### PR TITLE
Feature/batch upload

### PR DESCRIPTION
**Pull request by bjoerge**
This adds support for batch uploading through drag and drop to an array.

Its imlemented by a new infrastructure for _uploaders_, which is, in essence, a mapping from a mime type to a schema type, with a handler that converts a file to a schema type. Currently mappings for mime type `image/*` => schema type `image`, and `*/*` => `file` is implemented, but in the future we can expand this infrastructure to allow consumers to define their own uploaders.

### CODE REVIEW

**Comment by bjoerge** [comment]
Sorry about all the wip-commits, I'll squash all of these before merging.

File: packages/@sanity/form-builder/src/sanity/uploads/uploadImage.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import {CLEANUP_EVENT, INIT_EVENT, createUploadEvent} from './utils'
+import {createUploadId, getUploadEvents, scheduleUpload} from './uploadQueue'
+
+type Exif = {
+  orientation: OrientationId
+}
+
+const setInitialUploadState$ = Observable.of(INIT_EVENT)
+const unsetUploadState$ = Observable.of(CLEANUP_EVENT)
+
```
Comment (rexxars) [inline packages/@sanity/form-builder/src/sanity/uploads/uploadImage.js]: Should this be removed?

Comment (bjoerge) [reply packages/@sanity/form-builder/src/sanity/uploads/uploadImage.js]: Yeah, I forgot to delete it, just wanted to keep it around for a little while after testing concurrency yesterday. Thanks for the heads up 😅

**Comment by bjoerge** [comment]
@thomax made me aware I've managed to somehow break the input resolving for the block editor with these changes, so I'm looking into it now.

**Pull request merged**